### PR TITLE
Add array and dynamic offset types

### PR DESCRIPTION
**Pull request by cwfitzgerald**
What it says on the tin, adds ArrayMember and DynamicOffsetMember types which pad you T to 16 or 256 respectively.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -282,6 +284,47 @@ pub type DMat3 = DMat3x3;
...
+#[repr(C, align(16))]
+#[derive(Debug, Copy, Clone, Default, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct ArrayMember<T>(pub T);
+
+#[cfg(feature = "bytemuck")]
+unsafe impl<T: bytemuck::Zeroable> bytemuck::Zeroable for ArrayMember<T> {}
+#[cfg(feature = "bytemuck")]
+unsafe impl<T: bytemuck::Pod> bytemuck::Pod for ArrayMember<T> {}
+
+/// Pads a structure for use with dynamic offsets in wgpu.
```
Comment (LPGhatguy) [inline src/lib.rs]: It this just for wgpu? Is this also necessary for Vulkan or any other graphics APIs?

File: src/lib.rs [hunk 2]
```diff
@@ -282,6 +284,47 @@ pub type DMat3 = DMat3x3;
 /// Matrix of f64s with 4 columns and 4 rows. Alignment 32, size 128.
 pub type DMat4 = DMat4x4;
 
+/// Pads an element to be in an array in a shader.
+///
+/// All elements in arrays need to be aligned to 16s. This automatically aligns your types to 16s.
```
Comment (LPGhatguy) [inline src/lib.rs]:
```suggestion
/// All elements in arrays need to be aligned to 16 bytes. This automatically aligns your types to 16 bytes.
```

This wording is strange. Should we mention bytes here?

File: src/lib.rs [hunk 3]
```diff
@@ -282,6 +284,47 @@ pub type DMat3 = DMat3x3;
...
+pub struct ArrayMember<T>(pub T);
+
+#[cfg(feature = "bytemuck")]
+unsafe impl<T: bytemuck::Zeroable> bytemuck::Zeroable for ArrayMember<T> {}
+#[cfg(feature = "bytemuck")]
+unsafe impl<T: bytemuck::Pod> bytemuck::Pod for ArrayMember<T> {}
+
+/// Pads a structure for use with dynamic offsets in wgpu.
+///
+/// All dynamic offsets need to be aligned to 256s. This automatically aligns your types to 256s.
```
Comment (LPGhatguy) [inline src/lib.rs]:
```suggestion
/// All dynamic offsets need to be aligned to 256 bytes. This automatically aligns your types to 256 bytes.
```
Same suggestion as above.

**Review by LPGhatguy — COMMENTED** [review]
Small comments. Should we add an example for `DynamicOffsetMember` as well? It's pretty simple.

The one README example is getting a little bit big!

File: src/lib.rs (hunk 1 above)
Comment (cwfitzgerald) [reply src/lib.rs]: I know wgpu is the most restrictive, not sure about what all the others need. I'll just make it a generic "graphics apis".

**Review by LPGhatguy — APPROVED** [review]
Solid stuff!

**Pull request merged**